core/src/context.rs

```diff
@@ -184,6 +191,13 @@ impl Context {
             .as_ref()
             .expect("Observer channel not initialized")
     }
+
+    pub fn block_on<F: Future>(&self, future: F) -> <F as Future>::Output  {
+        match self.future_executor {
+            None => ThreadPool::new().expect("Default ThreadPool has to be spawnable"),
```

Comment:
So, when there is no executor, you are just spawning new ones into the void?

Reply:
Yeah, I've replaced a `panic!` with this. Should only happen in tests where we use a context without an instance.